spec/unit/provider/service/systemd_spec.rb

```diff
@@ -1,20 +1,19 @@
 require 'spec_helper'
 
-describe 'Puppet::Type::Service::Provider::Systemd', unless: Puppet::Util::Platform.jruby? do
-  let(:provider_class) { Puppet::Type.type(:service).provider(:systemd) }
+describe Puppet::Type.type(:service).provider(:systemd), unless: Puppet::Util::Platform.jruby? do
```

Comment:
As per https://relishapp.com/rspec/rspec-core/v/3-0/docs/subject/implicitly-defined-subject seems like setting describe like this it will also make an implicit subject available as the instance of the class in describe. Since creating a new provider might also evaluate facts we need to check if the subject is lazy initialized and if changing to this structure has any impact on tests performance and things that get called.

I see both `String class name` and `Class` approaches are used in Puppet for describing.

Reply:
Seems like the change between 5 and 6 for using `String` in `describe` was made with a reason: https://github.com/puppetlabs/puppet/commit/6994591e00f
We need to check if that still apply

Reply:
Brought back `provider_class` and made only the essential changes to fix the merge up asap.

Reply:
Jruby version could also be playing a role: 
`jruby-9.2.0.0` (at the [time](https://github.com/puppetlabs/puppet/blob/6994591e00f848ea730dc8d557ee455f2a032a5b/.travis.yml) of the commit mentioned) vs. `jruby-9.2.8.0` [(now)](https://github.com/puppetlabs/puppet/blob/master/.travis.yml)
But I think this investigation should be treated in a dedicated ticket and just move forward for now.